sklearnex/decomposition/pca.py

```diff
@@ -15,4 +15,313 @@
+    from sklearn.utils.validation import check_is_fitted
+
+    from daal4py.sklearn._utils import (
+        control_n_jobs,
+        run_with_n_jobs,
+        sklearn_check_version,
+    )
+
+    from .._device_offload import dispatch
+    from .._utils import PatchingConditionsChain
+
+    if sklearn_check_version("1.1") and not sklearn_check_version("1.2"):
+        from sklearn.utils import check_scalar
+
+    from sklearn.decomposition import PCA as sklearn_PCA
+
+    from onedal.decomposition import PCA as onedal_PCA
+
+    @control_n_jobs
+    class PCA(sklearn_PCA):
+        __doc__ = sklearn_PCA.__doc__
+
+        if sklearn_check_version("1.2"):
+            _parameter_constraints: dict = {**sklearn_PCA._parameter_constraints}
+
+        def __init__(
+            self,
+            n_components=None,
+            *,
+            copy=True,
+            whiten=False,
+            svd_solver="auto",
+            tol=0.0,
+            iterated_power="auto",
+            n_oversamples=10,
+            power_iteration_normalizer="auto",
+            random_state=None,
+        ):
+            self.n_components = n_components
+            self.copy = copy
+            self.whiten = whiten
+            self.svd_solver = svd_solver
+            self.tol = tol
+            self.iterated_power = iterated_power
+            self.n_oversamples = n_oversamples
+            self.power_iteration_normalizer = power_iteration_normalizer
+            self.random_state = random_state
+
+        def fit(self, X, y=None):
+            self._fit(X)
+            return self
+
+        def _fit(self, X):
+            if sklearn_check_version("1.2"):
+                self._validate_params()
+            elif sklearn_check_version("1.1"):
+                check_scalar(
+                    self.n_oversamples,
+                    "n_oversamples",
+                    min_val=1,
+                    target_type=numbers.Integral,
+                )
+
+            X = self._validate_data(
+                X,
+                dtype=[np.float64, np.float32],
+                ensure_2d=True,
+                copy=self.copy,
+            )
+
+            U, S, Vt = dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_PCA._fit,
+                },
+                X,
+            )
+            return U, S, Vt
+
+        @run_with_n_jobs
+        def _onedal_fit(self, X, queue=None):
+            if issparse(X):
+                raise TypeError(
+                    "PCA does not support sparse input. See "
+                    "TruncatedSVD for a possible alternative."
+                )
+
+            self._validate_n_components(self.n_components, X.shape[0], X.shape[1])
+
+            onedal_params = {
+                "n_components": self.n_components,
+                "is_deterministic": True,
+                "method": "cov",

```

Comment:
```suggestion
                "method": self.method,

```

Reply:
"cov" method is a oneDAL one, not from stock sklearn. Need to set it a hard coded.